### PR TITLE
Feature: Add 'Additional fields' FFM recommendation to form builder

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
## Description

This PR inserts the Additional fields panel inside the secondary sidebar on the primary block list. The panel displays the additional fields you can use in the form builder if you install the FFM add-on. A link is also displayed  which opens a new web page with additional information.The Additional fields panel will only render when FFM is NOT active and installed. The fields are not interactive, as such they have a "locked" appearance.

To limit the Additional panel to display only on the Primary block list, a Gutenberg selector was used to check for a selectedBlock. The primary block list is only visible when a block other than ```givewp/section```  has been selected.

## Visuals

Screen Shot 2023-08-23 at 3 02 02 PM

## Testing Instructions
- Add a new form.
- You will need to select a block and then open the secondary sidebar.
- Scroll to the bottom of the list to view the additional fields.
- Test the link.
- Install FormFieldManager
- View the form again and verify that it does not display.

- In the video you may notice the upgrade button text changes to a dark color after being clicked. This has since been fixed.
https://www.loom.com/share/9dc8307e041e4f1e97ddbf7d8a9a60b7?sid=60e41b30-dde1-4029-8418-3866b1a4955c

## Pre-review Checklist

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes update to unit tests to include new value for the view model [tests/Unit/ViewModels/FormBuilderViewModelTest.php](https://github.com/impress-org/givewp/pull/6897/files#diff-d1f7b58727a355c9bf21034aab9e1577105357b9765ac21ecbb350959c460056)
-   [x ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205360468064596

### CODE REVIEW

File: src/FormBuilder/resources/js/form-builder/src/promos/additionalFields/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                        'block-editor-inserter__panel-title block-editor-inserter__panel-additional-fields-header'
+                    }
+                >
+                    {__('Additional Fields', 'give')}
+                    <LockIcon />
+                </h2>
+            </div>
+
+            <div className="block-editor-inserter__panel-content">
+                <a
```
Comment (kjohnson) [inline src/FormBuilder/resources/js/form-builder/src/promos/additionalFields/index.tsx:24]: We should add `rel="noopener noreferrer"` for external links.

Reference: https://lw.slack.com/archives/C02QJ8F07B3/p1668803646241939?thread_ts=1668790561.688639&cid=C02QJ8F07B3

Comment (JoshuaHungDinh) [reply src/FormBuilder/resources/js/form-builder/src/promos/additionalFields/index.tsx:24]: Resolved: 2a9857e61a638d7873b224f9150d298d3dd22b7e

**Review by kjohnson — APPROVED** [review]
Ok, to QA!

**Review by rickalday — APPROVED** [review]
Passed QA tests

**Pull request merged**